src/magic.cpp

```diff
@@ -622,6 +628,11 @@ bool spell::is_valid_target( const tripoint &p ) const
     return valid;
 }
 
+bool spell::is_valid_effect_target( valid_target t ) const
```

Comment:
```suggestion
bool spell::is_valid_effect_target( const valid_target &t ) const
```

Reply:
valid_target is enum/int. Making it const ref is as pointless, as `const int& value`.

Reply:
mm, it is true that ref isn't that useful, but isn't const at least a good idea?

Reply:
but at this point it's just habit from all the stuff that's going on; if it's totally pointless then np